api/filters.go

```diff
@@ -557,6 +623,34 @@ func (api *API) isValidDatasetDimensions(w http.ResponseWriter, ctx context.Cont
 	return true
 }
 
+// validateCantabularDimensionExists checks that dimension exists in Cantabular by searching for it.
+// If the dimension doesn't exist, or couldn't be retrieved, an error is returned.
+func (api *API) validateCantabularDimensionExists(ctx context.Context, filterID, dimensionName string) error {
+	filter, err := api.store.GetFilter(ctx, filterID)
+	if err != nil {
+		return &badRequestErr{
+			errors.Wrap(err, "error retrieving filter"),
+		}
+	}
+
+	foundDimensions, err := api.ctblr.SearchDimensions(ctx, cantabular.SearchDimensionsRequest{
+		Dataset: filter.PopulationType,
+		Text:    dimensionName,
+	})
+
+	if err != nil {
```

Comment:
better to keep error checks immediately after generated (even if it was a multi-line call)